src/Microsoft.Extensions.FileProviders.Abstractions/NullFileProvider/NullDirectoryContents.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Extensions.FileProviders.Abstractions
+{
+    public class NullDirectoryContents : IDirectoryContents
+    {
+        public bool Exists => false;
+
+        public IEnumerator<IFileInfo> GetEnumerator()
+        {
+            yield break;
```

Comment:
Wouldn't it be better to use `Enumerable.Empty<IFileInfo>`? Much more declarative.